### PR TITLE
bt-30: Create Bottom tabs navigation

### LINKED ISSUE #30: Create Bottom tabs navigation
Add a new Navigation component with Bottom tabs. Each tab should lead to a separate screen (screens can be empty for now). 

Image

### PR DESCRIPTION
**Pull request by anogire**
I have made a separate AuthNavigation, which should include all pages for authorized users (maybe there will be more, not just those in the bottom tab).
So, app.tsx will probably have a switch to AuthNavigation or RootNavigation.

### CODE REVIEW

**Comment by likeri29** [comment]
I thought that you needed Icons for buttons and they are not on the development branch yet.

**Comment by anogire** [comment]
yes, i am waiting for icons.

**Review by nikita-remeslov — APPROVED** [review]
Looks good to me, but we need icons

**Comment by katerynakostikova** [comment]
So, I'd like to change the structure a bit. We should have a RootNavigator component, which will work like this:
```
               <RootStack.Navigator>
			{!isSignedIn ? (
						<RootStack.Screen
							name={AUTH_ROOT_ROUTE} // This Navigator should have auth-related screens 
							component={AuthNavigator}
						/>
					) : 
                                                   // This one will have bottom tabs 
						<RootStack.Screen name={MAIN_ROOT_ROUTE} component={MainRootNavigator} />
				</RootStack.Group>
			)}
		</RootStack.Navigator>
```
For now, you can manually change the value of `isSignedIn` 

Also, please provide screenshots of created navigation (with icons when they'll be merged)

File: mobile/src/app/app.tsx [hunk 1]
```diff
@@ -7,14 +7,18 @@ import { Provider as StoreProvider } from 'react-redux';
...
+} from '~/navigations/navigations';
 
 const App: FC = () => {
     return (
         <StoreProvider store={store.instance}>
             <GestureHandlerRootView style={globalStyles.flex1}>
                 <NavigationContainer>
-                    <RootNavigation />
+                    {/* <RootNavigation /> */}
+                    <AuthNavigation />
```
Comment (katerynakostikova) [inline mobile/src/app/app.tsx]: All related to my comment manipulations should be in RootNavigation component, in App we just leave RootNavigation

**Review by katerynakostikova — CHANGES_REQUESTED** [review]

**Comment by anogire** [comment]
1. If I understand correctly, we will also need MainNavigator for Drawer navigation.

2. Not sure if I need to separate generateTabBarIcon ...

3. If isSignedIn == true
Screenshot 2023-08-24 at 13 11 31

Screenshot 2023-08-24 at 13 11 37

Screenshot 2023-08-24 at 13 11 42


4.   If isSignedIn == false
Screenshot 2023-08-24 at 13 15 40

**Review by mskmee — APPROVED** [review]
Looks good, as for me)

**Comment by anogire** [comment]
Or, is Drawer navigation our "Sign Up page" that should be without bottom navigation?
Then, probably, a separate BottomTabNavigator is unnecessary and I can move it to the MainNavigator.

File: mobile/src/bundles/common/enums/navigation/root-screen-name.enum.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const RootScreenName = {
-    SIGN_UP: 'Sign Up',
-    SIGN_IN: 'Sign In',
+    AUTH_ROOT_ROUTE: 'Auth root route',
+    MAIN_ROOT_ROUTE: 'Main root route',
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/enums/navigation/root-screen-name.enum.ts]:
```suggestion
    AUTH_ROOT_ROUTE: 'Auth Root Route',
    MAIN_ROOT_ROUTE: 'Main Root Route',
```

File: mobile/src/navigations/bottom-tab/bottom-tab.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            <BottomTab.Screen
+                name={BottomTabScreenName.MAIL}
+                component={Mail}
+                options={{
+                    tabBarIcon: ({ color }) =>
+                        generateTabBarIcon({
+                            name: 'mail',
+                            color,
+                        }),
+                }}
```
Comment (katerynakostikova) [inline mobile/src/navigations/bottom-tab/bottom-tab.tsx]:
```suggestion
                options={{
                    tabBarIcon: ({ color }) =>
                       <Icon />
                }}
```

File: mobile/src/navigations/bottom-tab/generate-tab-bar-icon.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+type Properties = {
+    name: string;
+    color: string;
+};
+
+const generateTabBarIcon = ({ name, color }: Properties): React.ReactNode => {
+    const defaultSize = 24;
+
+    return <Icon name={name} size={defaultSize} color={color} />;
+};
```
Comment (katerynakostikova) [inline mobile/src/navigations/bottom-tab/generate-tab-bar-icon.tsx]: You can just return icon component in navigator

File: mobile/src/navigations/bottom-tab/styles.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import { Color } from '~/bundles/common/enums/enums';
+
+const bottomTabStyles: BottomTabNavigationOptions = {
+    headerShown: false,
+    tabBarActiveTintColor: '#FFFFFF',
+    tabBarInactiveTintColor: Color.TEXT2,
+    tabBarStyle: {
+        backgroundColor: Color.PRIMARY,
+        height: 95,
```
Comment (katerynakostikova) [inline mobile/src/navigations/bottom-tab/styles.ts]: Why 95?

File: mobile/src/navigations/main-navigator/main-navigator.tsx
Comment (katerynakostikova) [inline mobile/src/navigations/main-navigator/main-navigator.tsx:1]: We can remove this main navigator

File: mobile/src/navigations/root/root.tsx [hunk 6]
```diff
@@ -4,28 +4,37 @@ import {
...
-            <NativeStack.Screen
-                name={RootScreenName.SIGN_UP}
-                component={Auth}
-            />
-        </NativeStack.Navigator>
+        <RootStack.Navigator screenOptions={screenOptions}>
+            {isSignedIn ? (
+                <RootStack.Screen
+                    name={RootScreenName.MAIN_ROOT_ROUTE}
+                    component={MainNavigator}
```
Comment (katerynakostikova) [inline mobile/src/navigations/root/root.tsx]:
```suggestion
                    component={MainBottomTabNavigator}
```

File: mobile/src/navigations/bottom-tab/styles.ts (hunk 5 above)
Comment (anogire) [reply mobile/src/navigations/bottom-tab/styles.ts]: Yes, in Figma it is 97 ...

File: mobile/src/navigations/main-bottom-tab/styles.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import { Color } from '~/bundles/common/enums/enums';
+
+const bottomTabStyles: BottomTabNavigationOptions = {
+    headerShown: false,
+    tabBarActiveTintColor: '#FFFFFF',
+    tabBarInactiveTintColor: Color.TEXT2,
+    tabBarStyle: {
+        backgroundColor: Color.PRIMARY,
+        height: 97,
```
Comment (katerynakostikova) [inline mobile/src/navigations/main-bottom-tab/styles.ts]:
```suggestion
```
Let's make it 50 or just add some paddings. Let's avoid using some magical numbers

**Review by katerynakostikova — APPROVED** [review]

File: mobile/src/navigations/main-bottom-tab/styles.ts (hunk 7 above)
Comment (katerynakostikova) [reply mobile/src/navigations/main-bottom-tab/styles.ts]: For example on my phone 97 height looks too big, because I also have android bottom menu

**Comment by anogire** [comment]
Padding does not work correctly, only paddingTop or using `<View><Icon></View>`
So I suggest still using height

This is for 50
Screenshot 2023-08-25 at 11 08 08


This is for 60
Screenshot 2023-08-25 at 11 08 46

**Comment by katerynakostikova** [comment]
Let's use 60

**Review by katerynakostikova — APPROVED** [review]

**Comment by anogire** [comment]
I also like it better

**Pull request merged**